internal/blockchain/ethereum/quorum/geth_provider.go

```diff
@@ -0,0 +1,409 @@
+// Copyright © 2024 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quorum
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/hyperledger/firefly-cli/internal/blockchain/ethereum"
+	"github.com/hyperledger/firefly-cli/internal/blockchain/ethereum/connector"
+	"github.com/hyperledger/firefly-cli/internal/blockchain/ethereum/connector/ethconnect"
+	"github.com/hyperledger/firefly-cli/internal/blockchain/ethereum/connector/evmconnect"
+	"github.com/hyperledger/firefly-cli/internal/constants"
+	"github.com/hyperledger/firefly-cli/internal/docker"
+	"github.com/hyperledger/firefly-cli/internal/log"
+	"github.com/hyperledger/firefly-cli/pkg/types"
+)
+
+var gethImage = "quorumengineering/quorum:24.4"
+var tesseraImage = "quorumengineering/tessera:24.4"
+var exposedBlockchainPortMultiplier = 10
+
+// TODO: Probably randomize this and make it different per member?
+var keyPassword = "correcthorsebatterystaple"
+
+type GethProvider struct {
+	ctx       context.Context
+	stack     *types.Stack
+	connector connector.Connector
+}
+
+func NewGethProvider(ctx context.Context, stack *types.Stack) *GethProvider {
```

Comment:
Just a thought I know a lot of this code is inherited from the geth provider and that Quorum obviously includes a go-ethereum client but wondering if we should rename all of this to a `qurorum_provider` will leave up to you to think that through

Reply:
The amount of shared code is quite large, it feels the only difference is the tesseraEnabled checks

Reply:
I struggled with this initially, but due to the other consensus algorithms and very different start up parameters for Quorum (e.g. needing to pass in other cli args, having to know the other Quorum endpoints, needing to wait for Tessera to be up first before starting the binary etc.), there are sufficiently large deviations from geth_provider.

Also, geth_provider in geth starts a single geth client, while geth_provider in quorum starts multiple quorum and or tessera pairs. Each quorum client maintains it's own private state trie, so we can't reuse the code to spin up a single client.

Once we add in support for IBFT/QBFT, there will be deviations in the `genesis.go` file in the same package as well. 

Will rename to quorum_provider instead of geth_provider

Reply:
Thanks for the clarification